zipkin-collector/src/main/scala/com/twitter/zipkin/collector/BlockingItemQueue.scala

```diff
@@ -0,0 +1,36 @@
+package com.twitter.zipkin.collector
+
+import com.twitter.conversions.time._
+import com.twitter.finagle.stats.{DefaultStatsReceiver, StatsReceiver}
+import com.twitter.logging.Logger
+import com.twitter.util.{Duration, Future}
+
+/**
+ * This queue inherits from ItemQueue but instead of throwing an exception when its full it sleeps
+ * for a configurable amount of time until the span storer has enough time to consume records.
+ */
+class BlockingItemQueue [A, B](
+  maxSize: Int,
+  maxConcurrency: Int,
+  process: A => Future[B],
+  timeout: Duration = Duration.Top,
+  sleepOnFull: Duration = 30.seconds,
```

Comment:
This really depends on the application. I chose 30 because our application was seriously falling behind. I'll change it to 1s though. It sounds closer to what a lot of people will face.